src/Workspaces/Core/Portable/Serialization/SerializerService_Reference.cs

```diff
@@ -504,13 +525,26 @@ private static void WriteUnresolvedAnalyzerReferenceTo(AnalyzerReference referen
         }
         catch
         {
-            // we have a reference but the file the reference is pointing to
-            // might not actually exist on disk.
-            // in that case, rather than crashing, we will handle it gracefully.
+            // We have a reference but the file the reference is pointing to might not actually exist on disk. In that
+            // case, rather than crashing, we will handle it gracefully.
             return null;
         }
     }
 
+    private static Guid TryGetAnalyzerFileReferenceMvid(AnalyzerFileReference file)
+    {
+        try
+        {
+            return AssemblyUtilities.ReadMvid(file.FullPath);
+        }
+        catch
+        {
+            // We have a reference but the file the reference is pointing to might not actually exist on disk. In that
+            // case, rather than crashing, we will handle it gracefully.
+            return Guid.Empty;
```

Comment:
Another case is that a user has provided an analyzer file reference that isn't a valid managed assembly. That does happen from time to time. Would also trigger this code path.